lib/trails.js

```diff
@@ -27,6 +53,7 @@ module.exports = {
       log: { }
     }
     const envConfig = (config.env && config.env[process.env.NODE_ENV]) || { }
+    config = this.deepMerge((config || { }), (envConfig || { }))
```

Comment:
If you do this all line bellow are useless normally because they merge env for main and log witch is already done by this line